### PR TITLE
Add script for deploying a digital twin that matches the smart trailer use case

### PR DESCRIPTION
**Pull request by devkelley**
This PR adds a deployment script that creates and initializes an Azure Digital Twin with a Twin model that matches the Smart Trailer Use case. Updates and adds some documentation.

### CODE REVIEW

File: cloud_connectors/azure/digital_twins_connector/README.md [hunk 1]
```diff
@@ -36,6 +36,24 @@ chmod +x digital_twins_setup.sh
...
+```shell
+az login --use-device-code
+```
+
+1. Deploy Azure Digital Twins to your resource group.
+
+```shell
+cd {freyja-root-dir}/cloud_connectors/azure/scripts
+chmod +x digital_twins_setup_smart_trailer.sh
+./digital_twins_setup.sh -r myRG -l westus2 -d myADT
```
Comment (wilyle) [inline cloud_connectors/azure/digital_twins_connector/README.md:54]:
```suggestion
./digital_twins_setup.sh -r {myRG} -l westus2 -d {myADT}
```

maybe put the region in braces too?

File: cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    echo "Digital Twins instance '$digital_twin_name' already exists in resource group '$resource_group'"
+else
+    echo -e "\nCreating the Azure Digital Twins resource"
+    az dt create --dt-name "$digital_twin_name" --resource-group "$resource_group" --location "$location"
+fi
+
+# Assign the Digital Twins Data Owner role
+echo -e "\nAssigning the Azure Digital Twins Data Owner role"
+userObjectID=$(az ad signed-in-user show --query id -o tsv)
+az dt role-assignment create --dt-name "$digital_twin_name" --assignee "$userObjectID"  --role "Azure Digital Twins Data Owner"
```
Comment (wilyle) [inline cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh:73]: some variables are in pascal case and some are in snake case, let's pick one and be consistent. I'm not sure what's commonly used in bash scripts though

File: cloud_connectors/azure/digital_twins_connector/README.md [hunk 3]
```diff
@@ -36,6 +36,24 @@ chmod +x digital_twins_setup.sh
 ./digital_twins_setup.sh -r myRG -l westus2 -d myADT
 ```
 
+### Automated Azure Digital Twins Setup for Smart Trailer Example
+
+Before starting this section, please view [Prerequisites for Automated Deployment of Azure Resources](../README.md#prerequisites-for-automated-deployment-of-azure-resources).
```
Comment (ashbeitz) [inline cloud_connectors/azure/digital_twins_connector/README.md:41]: ... please read ...

File: cloud_connectors/azure/digital_twins_connector/README.md [hunk 4]
```diff
@@ -36,6 +36,24 @@ chmod +x digital_twins_setup.sh
...
+
+```shell
+az login --use-device-code
+```
+
+1. Deploy Azure Digital Twins to your resource group.
+
+```shell
+cd {freyja-root-dir}/cloud_connectors/azure/scripts
+chmod +x digital_twins_setup_smart_trailer.sh
```
Comment (ashbeitz) [inline cloud_connectors/azure/digital_twins_connector/README.md:53]: The chmod should not be necessary if the script in the repo has those settings,

File: cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+usage() {
+    echo "Usage: $0 [-r|--resource-group-name] <RESOURCE_GROUP_NAME> [-l|--location] <DEPLOYMENT_LOCATION> [-d|--digital-twins-name] <DIGITAL_TWINS_NAME>"
+    echo "Example:"
+    echo "  $0 -r myRG -l westus2 -d myADT"
+}
+
+# Parse command line arguments
+while [[ $# -gt 0 ]]
+do
+key="$1"
```
Comment (ashbeitz) [inline cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh:22]: Please consider using indentation for the while loop's body.

File: cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+case $key in
+    -r|--resource-group-name)
+    resource_group="$2"
```
Comment (ashbeitz) [inline cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh:26]: Please consider using indentation for each of the case clause's statements.

**Review by ashbeitz — COMMENTED** [review]
Some minor comments.

File: cloud_connectors/azure/digital_twins_connector/README.md (hunk 4 above)
Comment (devkelley) [reply cloud_connectors/azure/digital_twins_connector/README.md:53]: I think at least some of the scripts do not have those settings so want to leave this in to keep parity with other documentation.

File: cloud_connectors/azure/digital_twins_connector/README.md (hunk 1 above)
Comment (devkelley) [reply cloud_connectors/azure/digital_twins_connector/README.md]: Updated in this section and the section I copied the directions from

File: cloud_connectors/azure/digital_twins_connector/README.md (hunk 3 above)
Comment (devkelley) [reply cloud_connectors/azure/digital_twins_connector/README.md]: updated in this section and the one I copied from

File: cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh (hunk 5 above)
Comment (devkelley) [reply cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh]: Indented here and the script I used as base

File: cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh (hunk 6 above)
Comment (devkelley) [reply cloud_connectors/azure/scripts/digital_twins_setup_smart_trailer.sh]: Indented here and the script I used as base

File: cloud_connectors/azure/digital_twins_connector/README.md (hunk 4 above)
Comment (ashbeitz) [reply cloud_connectors/azure/digital_twins_connector/README.md:53]: Okay, leave it for now, but please try to eliminate this.

**Review by ashbeitz — APPROVED** [review]
Approved

**Review by wilyle — APPROVED** [review]

File: cloud_connectors/azure/digital_twins_connector/README.md [hunk 7]
```diff
@@ -33,7 +33,25 @@ az login --use-device-code
...
+./digital_twins_setup.sh -r {myRG} -l westus2 -d {myADT}
```
Comment (jorchiu) [inline cloud_connectors/azure/digital_twins_connector/README.md:54]:
```suggestion
./digital_twins_setup_smart_trailer.sh -r {myRG} -l westus2 -d {myADT}
```

**Pull request merged**